### DISCUSSION TITLE
Downgraded nextjs node version from 20 > 18 to run in vercel. when building project deployment in vercel, /lib64/libm.so.6: version `GLIBC_2.27' not found error

### DISCUSSION THREAD

**Discussion by rozzrr** [vercel/next.js#58133] [Help]
### Summary

i can run locally node v18 my nextjs project and build. when i deploy to vercel, i get the following output: 

<html>
<body>

npm ERR! code 1
--
12:30:09.838 | npm ERR! path /vercel/path0/node_modules/bufferutil
12:30:09.839 | npm ERR! command failed
12:30:09.840 | npm ERR! command sh -c node-gyp-build
12:30:09.840 | npm ERR! node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by node)
12:30:09.840 | npm ERR! node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by node)
12:30:09.841 |  
12:30:09.841 | npm ERR! A complete log of this run can be found in: /vercel/.npm/_logs/2023-11-07T12_29_52_076Z-debug-0.log
12:30:09.863 | Error: Command "npm install" exited with 1

</body>
</html>

i have downgraded from node20 to node18 for the LTS option as the most current offering that AWS/vercel supports. however i am stuck now with this error and im not sure how to fix

any help would be appreciated? 

thanks

**Comment by rozzrr** [comment]
ok so update: i compared my package.json from before and after specifying the build of node. i removed the node: 18.18.2 from the package json and outputted the node -v in the vercel build. vercel is using an older version of node 18 (18.17.1) so it must have been trying to get a newer version and it cannot. i have since deleted the nodejs version and using docker to maintain my local instance to match that of vercel.

**Reply by imsahil2001** [reply]
I found one workaround, you retain node version of your project to be v20 rather change the node version to 20 from the vercel project settings, and try to deploy it again.
I will update here if I get any issues.
